lib/main.go

```diff
@@ -225,6 +226,12 @@ func (obj *Main) Run() error {
 		}
 	}
 	Logf("working prefix is: %s", prefix)
+	// change to prefix directory to have a deterministic path where unix:// socket files are created
```

Comment:
I can't commit this block just yet until we've figured out the `chdir` plan as discussed, but I can commit everything else if you'd like to split this part out into a separate commit.